references/detection/train.py

```diff
@@ -36,12 +37,17 @@ def plot_samples(images, targets):
         img *= 255
         img = tf.cast(tf.clip_by_value(tf.round(img), 0, 255), dtype=tf.uint8).numpy()
 
-        target = np.zeros(img.shape[:2], dtype=bool)
+        target = np.zeros(img.shape[:2], np.uint8)
         boxes = targets[idx]['boxes'][np.logical_not(targets[idx]['flags'])]
         boxes[:, [0, 2]] = boxes[:, [0, 2]] * img.shape[1]
         boxes[:, [1, 3]] = boxes[:, [1, 3]] * img.shape[0]
         for box in boxes.round().astype(int):
-            target[box[1]: box[3] + 1, box[0]: box[2] + 1] = True
+            if rotation:
+                target[box[1]: box[3] + 1, box[0]: box[2] + 1] = 1
+            else:
+                box = cv2.boxPoints(((box[0], box[2]), (box[1], box[3]), box[4]))
+                box = np.int0(box)
+                cv2.fillPoly(target, [box], 1)
```

Comment:
let's cast the target back to bool afterwards (doing so explicitely tells the reader that each element can only have two values, and it's faster)